reference/src/glossary.md

```diff
@@ -1,5 +1,49 @@
 ## Glossary
 
+#### Aliasing
+
+(Please note: a full aliasing model for Rust has not yet been constructed, but
+at the moment we can give the following definition.)
+
+*Aliasing* is any time one pointer or reference points to a "span" of memory
+that overlaps with the span of another pointer or reference. A span of memory is
+similar to how a slice works: there's a base byte address as well as a length in
+bytes.
+
+Consider the following example:
+
+```rust
+fn main() {
+    let u: u64 = 7_u64;
+    let r: &u64 = &u;
+    let s: &[u8] = unsafe {
+        core::slice::from_raw_parts(&u as *const u64 as *const u8, 8)
+    };
+    let (head, tail) = s.split_first().unwrap();
+}
+```
+
+In this case, both `r` and `s` alias each other, since they both point to all of
+the bytes of `u`.
+
+However, `head` and `tail` do not alias each other: `head` points to the first
+byte of `u` and `tail` points to the other seven bytes of `u` after it. Also,
+both `head` and `tail` alias `s`.
+
+* The span length of `&T`, `&mut T`, `*const T`, or `*mut T` when `T` is
+  [`Sized`](https://doc.rust-lang.org/core/marker/trait.Sized.html) is
+  `size_of<T>()`.
+* When `T` is not `Sized` the span length is `size_of_val(t)`.
```

Comment:
- _cannot find `t` in this scope_ :P

  - also, **how can we get the size of the pointee of a raw pointer?** (`mem::size_of_val` takes a `&T`).

      - this point may deserve its own discussion

Reply:
yeah, it's own discussion, because that's a very tricky one, that T-libs has to handle.

Reply:
T-lang, rather. But for now I think we can just assume the thing is "forced" to reference type. `&*t` would be extra-explicit I guess.

Reply:
pointers are never automatically forced to references (very unsafe), so you'd have to use the `&*t` form in an unsafe block

Reply:
I know, but this is pseudo-Rust anyway. ;)

Reply:
One could use type ascription here maybe, to denote that the type of `t` is `T`, e.g.,  `size_of_val(t: T)`

Reply:
So with `size_of_val` it takes a `&T`, but you wouldn't write `size_of_val(&t)` when `t` is an unsized type since `t` is _already_ in reference form.